R/run_headless.R

```diff
@@ -34,53 +34,60 @@ run_headless <- function(
     )
   }
 
-  browser <- switch(
-    # only match on the first element... avoids having to use missing or pmatch
-    browser[1],
-    "chrome" = paste0(
+  if (
+    identical(browser[1], "chrome") &&
+    identical(system, "Windows")
+  ) {
+    browser <- function(url) {
+      cat("Opening browser with system2\n")
+      system2(
```

Comment:
I think you should use processx for this stuff -- better argument handling, control over the process, more consistent behavior across platforms, and it can automatically shut down the process when the R process exits (using `supervise=TRUE`).